ext/cuda/bilinearinterpolation.jl

```diff
@@ -0,0 +1,169 @@
+import ..ClimaInterpolations.BilinearInterpolation:
+    Bilinear,
+    set_source_range!,
+    interpolatebilinear!,
+    get_stencil_bilinear1d,
+    get_dims
+
+function interpolatebilinear!(
+    ftarget::CuArray{FT, N},
+    bilinear::B,
+    fsource::CuArray{FT, N},
+) where {FT, N, B}
+    @assert N ≥ 2
+    (leveldimssource..., nxs, nys) = size(fsource)
+    (leveldimstarget..., nxt, nyt) = size(ftarget)
+
+    nsourcex, nsourcey, ntargetx, ntargety = get_dims(bilinear)
+
+    @assert leveldimssource == leveldimstarget
+    @assert nxs == nsourcex && nys == nsourcey
+    @assert nxt == ntargetx && nyt == ntargety
+    # max threadgroups per grid is not officially specified by Metal
+    # It is possible this number is unlimited on newer Apple models (>m3)
+    maxxblocks = 1024
+    maxyblocks = 1024
+    # target n loops per dimension per threadgroup
+    targetloopsperblockdimx = 4
+    targetloopsperblockdimy = 4
```

Comment:
Some of the names in this section are hard to read without underscores. Maybe changes like this would help:
```suggestion
    target_loops_per_block_dimx = 4
    target_loops_per_block_dimy = 4
```